Modules/Core/Controllers/GetController.php

```
@@ -2,91 +2,113 @@
+        $ctype = $this->content_types[$file_ext] ?? $this->ctype_default;
+        $file_size = filesize($this->targetPath . $filename);
+        
         header('Expires: -1');
         header('Cache-Control: no-store, no-cache, must-revalidate, max-age=0');
         header('Pragma: no-cache');
         header('Content-Disposition: attachment; filename="' . $filename . '"');
         header('Content-Type: ' . $ctype);
         header('Content-Length: ' . $file_size);
+        
         readfile($this->targetPath . $filename);
     }
```

PHPMD (2.15.0)

getFile accesses the super-global variable $_SERVER. (undefined)

(Superglobals)